src/components/presentational/DiscreteScale/DiscreteScale.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { CSSProperties, useContext } from 'react';
+import { ColorDefinition, resolveColor } from '../../../helpers';
+import { LayoutContext } from '../Layout';
+import './DiscreteScale.css';
+import UnstyledButton from "../UnstyledButton";
+
+export interface DiscreteScaleProps {
+    tickCount: number;
+    minLabel?: string;
+    maxLabel?: string;
+    value?: number;
+    onChange?: (value: number) => void;
+    sliderColor?: ColorDefinition;
+    innerRef?: React.RefObject<HTMLDivElement>;
+}
+
+export default function (props: DiscreteScaleProps) {
+    const layoutContext = useContext(LayoutContext);
+
+    const values = [...Array(props.tickCount).keys()];
+
+    let selectedValue = props.value ?? 0;
+    if (selectedValue < 0) selectedValue = 0;
+    if (selectedValue >= props.tickCount) selectedValue = props.tickCount - 1;
+
+    let sliderColor = props.sliderColor ? resolveColor(layoutContext.colorScheme, props.sliderColor) : 'var(--mdhui-color-primary)';
+    let sliderStop = `calc((((100% - 16px) / ${(props.tickCount - 1)}) * ${selectedValue}) + 8px)`;
+    let sliderStyle = {
+        background: `linear-gradient(to right, ${sliderColor} 0%, ${sliderColor} ${sliderStop}, var(--mdhui-background-color-2) ${sliderStop}, var(--mdhui-background-color-2) 100%)`
+    } as CSSProperties;
+
+    let ticksStyle = {
+        gridTemplateColumns: values.slice(0, -1).map(v => '1fr').join(' ')
+    } as CSSProperties;
+
+    let buttonStyle = {
+        background: sliderColor
+    } as CSSProperties;
+
+    const onClick = (value: number) => {
+        if (props.onChange) {
+            props.onChange(value);
+        }
+    };
+
+    return <div className="mdhui-discrete-scale" ref={props.innerRef}>
```

Comment:
Not quite sure what where the best place to adjust this is, but I think the current layout is going to almost always cause horizontal scroll bars.  Both ends of the chart extend out past where they visually do because of the click targets:
![image](https://github.com/CareEvolution/MyDataHelpsUI/assets/5473619/5e8a0fc6-0108-4c29-8e92-131852956db8)

I think that adding `overflow: hidden` to the `mdhui-discrete-scale` class is sufficient to fix it.  It does mean that the first and last click target still overflow the bar and you can click slightly *outside* of the bar to select them.  But - honestly I think that's probably fine.  The other alternative would be to do some `first-child`/`last-child` variations to make the `mdhui-discrete-scale-ticks-section-button` smaller.  But that's probably more work and not worth it.

Reply:
Good catch.  I've fixed the issue with the horizontal scroll.  The touch targets are now contained within the bar.